scripts/demangle.py

```diff
@@ -97,50 +99,81 @@ def main(proj_name: str, proj_language: str = "c"):
             file_func_name = ""
             for file_func_name_ in data[cnt]:
                 file_func_name = file_func_name_
-            # if split not matched, need to add a check
+            # If split not matched, need to add a check
             splited_file_func_name = file_func_name.split(FILE_FUNC_DELIM)
             file_path, mangle_func_name = (
                 splited_file_func_name[0],
                 splited_file_func_name[1],
             )
 
-            # if this is c file then the mangle does not exist
-            try:
-                demangle_func_name = cpp_demangle.demangle(mangle_func_name)
-            except ValueError:
-                demangle_func_name = mangle_func_name
-            if demangle_func_name == None:
-                warning(f" {mangle_func_name} demangle incorrect or unable to demangle")
-                data[cnt][
-                    file_func_name
-                ] = f" {mangle_func_name}demangle incorrect or unable to demangle"
-            else:
-                func_name = demangle_func_name.split("(")[0]
-                # Get code path from local
-                code_path = get_source_code_path(file_path, output_path)
-                if code_path == None:
-                    warning(f"Get source code path error {file_path}")
+            # Get code path from local
+            code_path = get_source_code_path(file_path, output_path)
+            if code_path == None:
+                warning(f"Get source code path error {file_path}")
+                data[cnt][file_func_name] = {
+                    "code": "Path Error",
+                    "data": data[cnt][file_func_name],
+                }
+                continue
+
+            # Get function content
+            temp_proj_language = proj_language
```

Comment:
why is a temp variable needed here?

Reply:
Some C++ projects have some C source code so we can't just use the project's language and we have to look at the file extension to determine which function to use to fetch the source code. The temp variable is just for reseting `proj_language` back to its original value.

Reply:
please put this comment on the code